### PR TITLE
Spark doris connector http v2 authentication fails, and HTTP v2 interface returns json nesting problem

### PR DESCRIPTION
**Pull request by hf200012**
1. Deal with the problem of inconsistent data format returned by http v1 and v2
2. Deal with user authentication failure

### CODE REVIEW

File: extension/spark-doris-connector/src/test/scala/org/apache/doris/spark/sql/SparkDorisConnector.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        val sc = new SparkContext(sparkConf)
+        sc.setLogLevel("DEBUG")
+        import org.apache.doris.spark._
+        val dorisSparkRDD = sc.dorisRDD(
+//            tableIdentifier = Some("ods.ods_pos_pro_shop_delta"),
+            tableIdentifier = Some("ods.ods_t_pro_dish_list_detail_test_demo"),
+//            query = Some("select * from ods.ods_t_pro_dish_list_detail_test_demo limit 1"),
+            cfg = Some(Map(
+//                "doris.fenodes" -> "10.220.146.10:8030",
+                "doris.fenodes" -> "10.220.147.155:8030",
```
Comment (morningman) [inline extension/spark-doris-connector/src/test/scala/org/apache/doris/spark/sql/SparkDorisConnector.scala:20]: hide the ip

File: extension/spark-doris-connector/src/main/java/org/apache/doris/spark/rest/RestService.java [hunk 2]
```diff
@@ -32,17 +32,15 @@
...
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (morningman) [inline extension/spark-doris-connector/src/main/java/org/apache/doris/spark/rest/RestService.java:41]: Better not using `import *`

File: extension/spark-doris-connector/src/test/scala/org/apache/doris/spark/sql/SparkDorisConnector.scala (hunk 1 above)
Comment (hf200012) [reply extension/spark-doris-connector/src/test/scala/org/apache/doris/spark/sql/SparkDorisConnector.scala:20]: ok

File: extension/spark-doris-connector/src/main/java/org/apache/doris/spark/rest/RestService.java (hunk 2 above)
Comment (hf200012) [reply extension/spark-doris-connector/src/main/java/org/apache/doris/spark/rest/RestService.java:41]: OK

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**